### PR TITLE
4034-move-tcreport-site-generator-into-repo

### LINKED ISSUE #4034: Copy tcreport somewhere else

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolve #4034.

- Copied the contents of https://github.com/GCTC-NTGC/tc-report into this repo. The static content now lives at `/tc-report/_site`, instead of just `/tc-report`.
- Updated the htaccess files so tc-report content continues to be served correctly. Ran into some issues at first (first an infinite loop of apache rewrites, then folder names not serving correctly) but its working now.
- Slight modifications to README files about tc-report subfolder.
- Added a test to confirm I didn't break the /talent-cloud and /talent-cloud/report pages.

### CODE REVIEW

**Review by petertgiles — DISMISSED** [review]
Looks great!  I tested locally and on the dev server.  A few things to consider below.

File: tc-report/.vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  // See https://go.microsoft.com/fwlink/?LinkId=733558
+  // for the documentation about the tasks.json format
+  "version": "2.0.0",
+  "tasks": [
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply tc-report/.vscode/tasks.json:5]: Yep, good call.

File: tc-report/_site/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "url": "git+https://github.com/GCTC-NTGC/gc-digital-talent.git"
+  },
+  "author": "Josh Beveridge",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/GCTC-NTGC/gc-digital-talent/issues"
+  },
+  "homepage": "https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/tc-report/README.md",
+  "dependencies": {
+    "@hydrogen-design-system/hydrogen.css": "^1.1.12"
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply tc-report/_site/package.json:21]: I don't think we want to bother upgrading this, unless @JoshBeveridge really wants to. How do we ignore this on dependabot?

File: tc-report/_site/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "version": "1.0.0",
+  "description": "A static site that offers a web version of the Talent Cloud results report.",
+  "main": "index.js",
+  "scripts": {
+    "h2-compile": "npx h2-compile",
+    "h2-build": "npx h2-build"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/GCTC-NTGC/gc-digital-talent.git"
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply tc-report/_site/package.json:12]: I already did - do you think I should point it back to the original repo?

File: tc-report/_site/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (petertgiles) [reply tc-report/_site/package.json:21]: Ah, I double checked and it is set to ignore tc-report already. https://github.com/GCTC-NTGC/gc-digital-talent/blob/03edccb2c70a9cc56f4bf7f06b4860fd5c3ca3ef/.github/dependabot.yml#L35

File: tc-report/_site/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (petertgiles) [reply tc-report/_site/package.json:12]: Woops, you're right.  I must have been looking at the old links.  Yeah, this is good.

**Comment by tristan-orourke** [comment]
@petertgiles could you please take another look at my more recent commits?

**Comment by petertgiles** [comment]
The tasks work well and it looks nice locally but deployed it seems there are some routing issues.  It's on dev right now if you want to review:
image

**Comment by tristan-orourke** [comment]
Thanks @petertgiles! I fixed the issue. It reminded me that it would be nice to avoid having two htaccess files at all, and use the same one locally and in azure.

**Review by petertgiles — APPROVED** [review]
Looks great!  Good catch on the assets path.

**Pull request merged**